gems/pending/spec/postgres_ha_admin/failover_monitor_spec.rb

```diff
@@ -0,0 +1,141 @@
+describe PostgresHaAdmin::FailoverMonitor do
+  let(:db_yml) do
+    yml = double('DatabaseYml')
+    allow(yml).to receive(:pg_params_from_database_yml).and_return(:host => 'host.example.com', :user => 'root')
+    yml
+  end
+  let(:failover_db) { double('FailoverDatabases') }
+  let(:connection) do
+    conn = double("PGConnection")
+    allow(conn).to receive(:finish)
+    conn
+  end
+
+  let(:failover_monitor) do
+    expect(PostgresHaAdmin::DatabaseYml).to receive(:new).and_return(db_yml)
+    expect(PostgresHaAdmin::FailoverDatabases).to receive(:new).and_return(failover_db)
+    failover_instance = described_class.new('', '', @logger_file.path, 'test')
+    logger = double('Logger')
+    allow(logger).to receive(:info)
+    allow(logger).to receive(:error)
+    failover_instance.instance_variable_set("@logger", logger)
+    failover_instance
+  end
+
+  let(:linux_admin) do
+    linux_adm = double('LinuxAdmin')
+    allow(LinuxAdmin::Service).to receive(:new).and_return(linux_adm)
+    linux_adm
+  end
+
+  before do
+    @logger_file = Tempfile.new('ha_admin.log')
+  end
+
+  after do
+    @logger_file.close(true)
+  end
+
+  describe "#monitor" do
+    context "primary database is accessable" do
+      before do
+        allow(PG::Connection).to receive(:open).and_return(connection)
+      end
+
+      it "updates 'failover_databases.yml'" do
+        expect(failover_db).to receive(:update_failover_yml)
+        failover_monitor.monitor
+      end
+
+      it "does not stop evm server and does not execute failover" do
+        expect(failover_db).to receive(:update_failover_yml)
+        expect(linux_admin).not_to receive(:stop)
+        expect(linux_admin).not_to receive(:restart)
+        expect(failover_monitor).not_to receive(:execute_failover)
+
+        failover_monitor.monitor
+      end
+    end
+
+    context "primary database is not accessable" do
+      before do
+        allow(PG::Connection).to receive(:open).and_return(nil, connection, connection)
+      end
+
+      it "stop evm server(if it is running) before failover attempt" do
+        expect(linux_admin).to receive(:stop).ordered
+        expect(failover_monitor).to receive(:execute_failover).ordered
+
+        failover_monitor.monitor
+      end
+
+      it "does not update 'database.yml' and 'failover_databases.yml' if all standby DBs are in recovery mode" do
+        expect(failover_db).to receive(:active_databases).and_return(active_databases_list)
+        expect(failover_db).not_to receive(:update_database_yml)
+        expect(db_yml).not_to receive(:update_database_yml)
+        expect(linux_admin).not_to receive(:restart)
+
+        expect(failover_monitor).to receive(:stop_evmserverd).ordered
+        expect(PostgresAdmin).to receive(:database_in_recovery?).and_return(true, true, true).ordered
+
+        stub_const("PostgresHaAdmin::FailoverMonitor::FAILOVER_ATTEMPTS", 1)
+        stub_const("PostgresHaAdmin::FailoverMonitor::FAILOVER_CHECK_FREQUENCY", 1)
```

Comment:
If we are always doing this can it be somewhere common? Maybe in a helper method or the before hook if this will always be the case for tests in this context?